### ISSUE TITLE
Should cnn module in conformer be masked in dynamic chunk training?

### ISSUE THREAD

**Issue by cindycindyhi** [TeaPoly/Conformer-Athena#1]
In dynamic chunk training,  depthwise_conv1d and batchnorm in cnn_module will look ahead frames without mask.

**Comment by TeaPoly** [comment]
No need dynamic mask for  `depthwise_conv1d` and `batchnorm`, it's just for `attention` operation.
But you can change `depthwise_conv1d` to causal `depthwise_conv1d` if your model is working at streaming mode.

**Comment by cindycindyhi** [comment]
> No need dynamic mask for `depthwise_conv1d` and `batchnorm`, it's just for `attention` operation.
> But you can change `depthwise_conv1d` to causal `depthwise_conv1d` if your model is working at streaming mode.

Sorry, I can't understand why use `depthwise_conv1d` in training and `causal depthwise_conv1d` in inference.
If `depthwise_conv1d` kernel size is 7, it will look ahead 3 frames in training and the frames will accumulate frames with layers. `Attention` in higher layers will get more information from future.
In inference, if I use `causal depthwise_conv1d`, does performance drop a lot if training and inference don't match?

**Comment by TeaPoly** [comment]
> > No need dynamic mask for `depthwise_conv1d` and `batchnorm`, it's just for `attention` operation.
> > But you can change `depthwise_conv1d` to causal `depthwise_conv1d` if your model is working at streaming mode.
> 
> Sorry, I can't understand why use `depthwise_conv1d` in training and `causal depthwise_conv1d` in inference. If `depthwise_conv1d` kernel size is 7, it will look ahead 3 frames in training and the frames will accumulate frames with layers. `Attention` in higher layers will get more information from future. In inference, if I use `causal depthwise_conv1d`, does performance drop a lot if training and inference don't match?

I mean you can modify the training code from `depthwise_conv1d` to `causal depthwise_conv1d` by yourself.

**Comment by cindycindyhi** [comment]
> 

Get it. Thanks~

**Comment by cindycindyhi** [comment]
> > > No need dynamic mask for `depthwise_conv1d` and `batchnorm`, it's just for `attention` operation.
> > > But you can change `depthwise_conv1d` to causal `depthwise_conv1d` if your model is working at streaming mode.
> > 
> > 
> > Sorry, I can't understand why use `depthwise_conv1d` in training and `causal depthwise_conv1d` in inference. If `depthwise_conv1d` kernel size is 7, it will look ahead 3 frames in training and the frames will accumulate frames with layers. `Attention` in higher layers will get more information from future. In inference, if I use `causal depthwise_conv1d`, does performance drop a lot if training and inference don't match?
> 
> I mean you can modify the training code from `depthwise_conv1d` to `causal depthwise_conv1d` by yourself.


Get it, thanks~

**Comment by TeaPoly** [comment]
> > > > No need dynamic mask for `depthwise_conv1d` and `batchnorm`, it's just for `attention` operation.
> > > > But you can change `depthwise_conv1d` to causal `depthwise_conv1d` if your model is working at streaming mode.
> > > 
> > > 
> > > Sorry, I can't understand why use `depthwise_conv1d` in training and `causal depthwise_conv1d` in inference. If `depthwise_conv1d` kernel size is 7, it will look ahead 3 frames in training and the frames will accumulate frames with layers. `Attention` in higher layers will get more information from future. In inference, if I use `causal depthwise_conv1d`, does performance drop a lot if training and inference don't match?
> > 
> > 
> > I mean you can modify the training code from `depthwise_conv1d` to `causal depthwise_conv1d` by yourself.
> 
> Get it, thanks~

It looks like only minor modifications are needed. [Support causal depth wise conv1d.](https://github.com/TeaPoly/Conformer-Athena/commit/1fd896b86785e56618709f660f5c092c8ac29cac)
